htscodecs/rANS_static4x16pr.c

```diff
@@ -1147,18 +1080,26 @@ unsigned char *rans_compress_to_4x16(unsigned char *in,  unsigned int in_size,
 	if (!(out = malloc(*out_size)))
 	    return NULL;
     }
+    if (*out_size == 0)
+	return NULL;
```

Comment:
This test should happen before trying to allocate `out`.  Also, `out` could be leaked later if allocated above and one of a number of other allocations fails.